activity_calendar/models.py

```diff
@@ -85,3 +288,93 @@ def clean_fields(self, exclude=None):
         if errors:
             raise ValidationError(errors)
         
+
+
+class ActivitySlot(models.Model):
+    title = models.CharField(max_length=255)
+    description = models.TextField(blank=True)
+    location = models.CharField(max_length=255, blank=True, null=True,
+        help_text="If left empty, matches location with activity")
+    start_date = models.DateTimeField(blank=True, null=True,
+        help_text="If left empty, matches start date with activity")
+    end_date = models.DateTimeField(blank=True, null=True,
+        help_text="If left empty, matches end date with activity")
+    
+    # User that created the slot (or one that's in the slot if the original owner is no longer in the slot)
+    owner = models.ForeignKey(User, on_delete=models.CASCADE, blank=True, null=True)
```

Comment:
So if the owner of the slot deletes his account his past and future slots are deleted as well? Why not set blank?

Reply:
Good catch!